cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -231,35 +231,55 @@ func gitIndex(c gitIndexConfig, o *indexArgs) error {
 		return nil
 	}
 
-	err = fetch(o.Branches)
-	if err != nil {
-		return err
-	}
-
-	if o.UseDelta {
-		// Try fetching prior commits for delta builds
-		// If we're unable to fetch prior commits, we continue anyway
-		// knowing that zoekt-git-index will fall back to a "full" normal build
+	fetchPriorAndLatestCommits := func() error {
```

Comment:
@keegancsmith 

I realized the control flow was slightly more complicated that just whether `o.UseDelta` is set. We also need to consider whether not the repository has any prior shards on disk (so that we can figure out what the old commit is). 

I didn't like how deeply independent the control flow was becoming because of this. So, I tried refactoring this with a different approach (using functions) to see if this reads better. I'm not enamored with the result, but let me know what you think.